evalml/pipelines/time_series.py

```diff
@@ -0,0 +1,82 @@
+import pandas as pd
+
+from evalml.objectives import get_objective
+from evalml.pipelines.regression_pipeline import RegressionPipeline
+from evalml.problem_types import ProblemTypes
+from evalml.utils.gen_utils import any_values_are_nan, drop_nan, pad_with_nans
+
+
+class TimeSeriesRegressionPipeline(RegressionPipeline):
+    problem_type = ProblemTypes.TIME_SERIES_REGRESSION
+
+    def __init__(self, parameters, gap, max_delay, random_state=0):
+        super().__init__(parameters, random_state)
+        self.gap = gap
+        self.max_delay = max_delay
+
+    def fit(self, X, y=None):
+        """Fit a time series regression pipeline.
+
+        Arguments:
+            X (ww.DataTable, pd.DataFrame or np.ndarray): The input training data of shape [n_samples, n_features]
+            y (ww.DataColumn, pd.Series, np.ndarray): The target training targets of length [n_samples]
+
+        Returns:
+            self
+        """
+        if y is None:
+            assert isinstance(X, pd.Series), "When modeling a single time series, the data must be a series."
+        if not isinstance(X, pd.DataFrame):
+            X = pd.DataFrame(X)
+
+        if y is None:
+            X_t, _ = self._compute_features_during_fit(X, y=None)
+            y = X.squeeze()
+        else:
+            y = pd.Series(y)
+            X_t, _ = self._compute_features_during_fit(X, y)
+
+        y_shifted = y.shift(-self.gap)
+        X_t, y_shifted = drop_nan(X_t, y_shifted)
+        self.estimator.fit(X_t, y_shifted)
+        return self
+
+    def predict(self, X, y=None, objective=None):
+        """Make predictions using selected features.
+
+        Arguments:
+            X (ww.DataTable, pd.DataFrame, or np.ndarray): Data of shape [n_samples, n_features]
+            y (ww.DataColumn, pd.Series, np.ndarray, None): The target training targets of length [n_samples]
+            objective (Object or string): The objective to use to make predictions
+
+        Returns:
+            pd.Series: Predicted values.
+        """
+        features = self.compute_estimator_features(X, y)
+        predictions = self.estimator.predict(features.dropna(axis=0, how="any"))
```

Comment:
Should we use the `drop_nan` helper here too? Maybe I'm just confused about the names.

Reply:
Maybe I should rename the `drop_nan` helper `keep_non_nan_rows`. Since the target and features will have NaNs in different places in `score`, we need a way for selecting the rows that are not NaN in both of them. As far as I know, there isn't a pandas method that does this. 

In this case, we just need to drop the NaN rows from a dataframe and pandas has a built-in for it.

Reply:
Yep I like that name. Right, I don't think there's a convenient builtin in pandas for that particular use-case. I've always either used `dropna` or `isnan`.

Reply:
Great! I changed the name